### PR TITLE
BUG: unstack with object dtype of tzaware timestamps

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.3.0.rst [hunk 1]
```diff
@@ -1095,6 +1095,7 @@ Reshaping
 - Bug in :func:`to_datetime` raising an error when the input sequence contained unhashable items (:issue:`39756`)
 - Bug in :meth:`Series.explode` preserving the index when ``ignore_index`` was ``True`` and values were scalars (:issue:`40487`)
 - Bug in :func:`to_datetime` raising a ``ValueError`` when :class:`Series` contains ``None`` and ``NaT`` and has more than 50 elements (:issue:`39882`)
+- Bug in :meth:`Series.unstack` and :meth:`DataFrame.unstack` with object-dtype values containing timezone-aware datetime objects incorrectly raising ``TypeError`` (:issue:`41875`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.3.0.rst:1098]: did we not have a test that changed for this?

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v1.3.0.rst (hunk 1 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v1.3.0.rst:1098]: we recently changed the test for period and interval.  This makes the same change to that test for dt64tz

**Comment by jreback** [comment]
this is ok for 1.3, can you rebase

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
@meeseeksdev backport 1.3.x

**Comment by pcorpet** [comment]
It seems that it's not fixed for the case where mask_all is true. I'll send another PR
https://github.com/pandas-dev/pandas/blob/42082a80d0c3b3483f08b887271b1b08c66551e1/pandas/core/reshape/reshape.py#L294

**Pull request merged**